File: app/src/main/java/eungju/rxcurrencyconverter/ApplicationComponent.java
package eungju.rxcurrencyconverter;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {
    void inject(CurrencyApplication application);

    void inject(CurrencyConverterActivity activity);

    class Builder {
        public static ApplicationComponent build(CurrencyApplication app) {
            return DaggerApplicationComponent.builder()
                    .applicationModule(new ApplicationModule(app))
                    .build();
        }
    }
}

File: app/src/main/java/eungju/rxcurrencyconverter/ApplicationComponent.kt
package eungju.rxcurrencyconverter

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {
    fun inject(application: CurrencyApplication)

    fun inject(activity: CurrencyConverterActivity)

    object Builder {
        fun build(app: CurrencyApplication): ApplicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(app))
                .build()
    }
}
